syft/pkg/cataloger/java/archive_parser.go

```diff
@@ -135,6 +137,7 @@ func (j *archiveParser) parse() ([]*pkg.Package, []artifact.Relationship, error)
 // discoverMainPackage parses the root Java manifest used as the parent package to all discovered nested packages.
 func (j *archiveParser) discoverMainPackage() (*pkg.Package, error) {
 	// search and parse java manifest files
+	// TODO: do we want to prefer or check for pom files over manifest here?
```

Comment:
I'd proffer that we want to check both -- pom files may have a parent so the file itself won't necessarily have all the right info in it, but it _is_ evidence. Also, this doesn't seem pertinent to the change in this PR?

Reply:
This is pertinent to the change in this PR since pom files are where we could potentially get data that matches with the maven upstream records. The reason we are adding the digests is so that grype can eventually do a lookup. I noticed we didn't have a preference for the main package of manifest vs pom so wanted to ask the question.